flake.nix

```diff
@@ -0,0 +1,131 @@
+        import nixpkgs {
+          inherit system;
+          overlays = [ self.overlay ];
+        });
+
+    in {
+
+      overlay = final: prev: {
+        bitcoin-transactions =
+          let inherit (final) stdenv nodejs lib nodePackages;
+          in stdenv.mkDerivation {
+            name = "bitcoin-transactions";
+
+            src = bitcoin-transactions-src;
+
+            buildInputs = [ nodejs ];
+
+            dontBuild = true;
+
+            installPhase = ''
+              mkdir -p $out/
+              cp -r ./* $out/
+            '';
+
+            meta = {
+              homepage = "https://peersm.com/wallet";
+              description =
+                "Javascript implementation of the Bitcoin protocol for any Bitcoin based coins";
+              license = lib.licenses.mit;
+            };
+          };
+      };
+
+      # Provide some binary packages for selected system types.
+      packages = forAllSystems
+        (system: { inherit (nixpkgsFor.${system}) bitcoin-transactions; });
+
+      # These apps load the package as a standalone offline webapp
+      # For these to work the XDG browser variable needs to be set to something that exists
+      # To test run `xdg-settings get default-web-browser` if this fails
+      # prepend the nix run command with your installed browser with
+      # BROWSER=<installed browser> nix run ...
+      apps = forAllSystems (system: {
+        wallet = let pkgs = nixpkgsFor.${system};
+        in {
+          type = "app";
+          program = "${
+              pkgs.writeShellScriptBin "bitcoin-transactions" ''
+                ${pkgs.xdg-utils}/bin/xdg-open ${pkgs.bitcoin-transactions}/html/wallet.html
+              ''
+            }/bin/bitcoin-transactions";
+        };
+        # This shows the Javascript unminified and readable
+        wallet-clear = let pkgs = nixpkgsFor.${system};
+        in {
+          type = "app";
+          program = "${
+              pkgs.writeShellScriptBin "bitcoin-transactions" ''
+                ${pkgs.xdg-utils}/bin/xdg-open ${pkgs.bitcoin-transactions}/html/walletclear.html
+              ''
+            }/bin/bitcoin-transactions";
+        };
+      });
+
+      defaultPackage =
+        forAllSystems (system: self.packages.${system}.bitcoin-transactions);
+
+      devShell = forAllSystems (system:
+        let
+          pkgs = nixpkgsFor.${system};
+          inherit (pkgs) mkShell nodePackages;
+        in mkShell {
+          buildInputs = [ nodePackages.browserify nodePackages.terser ];
+          inputsFrom = builtins.attrValues self.packages.${system};
+        });
+
+      # Tests run by 'nix flake check' and by Hydra.
+      checks = forAllSystems (system: {
+        inherit (self.packages.${system}) bitcoin-transactions;
+
+        # Additional tests, if applicable.
+        test = with nixpkgsFor.${system};
+          stdenv.mkDerivation {
+            name = "bitcoin-transactions-test-${version}";
```

Comment:
I think this variable is missing

Reply:
Ah yeah I removed this as there is no clear versioning on the original project (forget to remove it in this test section) 

What are your thoughts on versioning for this package? Is it needed or do we just go for the last time the repo was last modified?

Reply:
Yeah, doesn't look like there is a release. Last time the repo was modified makes sense on my end.